tests/test_sample_size/test_variables.py

```diff
@@ -0,0 +1,121 @@
+import unittest
+from unittest.mock import patch
+
+from statsmodels.stats.power import NormalIndPower
+from statsmodels.stats.power import TTestIndPower
+
+from sample_size.metrics import BooleanMetric
+from sample_size.metrics import NumericMetric
+from sample_size.metrics import RatioMetric
+
+
+class BooleanMetricTestCase(unittest.TestCase):
+    def setUp(self):
+        self.DEFAULT_MDE = 0.01
+        self.DEFAULT_PROBABILITY = 0.05
+        self.DEFAULT_MOCK_VALUE = 99
+
+    @patch("sample_size.metrics.BooleanMetric._get_probability")
+    @patch("sample_size.metrics.BooleanMetric._get_variance")
+    def test_boolean_metric_constructor_sets_params(self, mock_get_variance, mock_get_probability):
+        mock_get_variance.return_value = self.DEFAULT_MOCK_VALUE
+        mock_get_probability.return_value = self.DEFAULT_PROBABILITY
+        boolean = BooleanMetric(self.DEFAULT_PROBABILITY, self.DEFAULT_MDE)
+
+        mock_get_variance.assert_called_once()
+        mock_get_probability.assert_called_once_with(self.DEFAULT_PROBABILITY)
+        self.assertEqual(boolean.probability, self.DEFAULT_PROBABILITY)
+        self.assertEqual(boolean.variance, self.DEFAULT_MOCK_VALUE)
+        self.assertEqual(boolean.mde, self.DEFAULT_MDE)
+        self.assertEqual(boolean.default_power_analysis_type, NormalIndPower)
+
+    def test_boolean_metric_get_variance(self):
+        boolean = BooleanMetric(self.DEFAULT_PROBABILITY, self.DEFAULT_MDE)
+        variance = boolean._get_variance()
+
+        self.assertEqual(variance, 0.0475)
+
+    def test_boolean_metric_get_probability(self):
+        probability = BooleanMetric._get_probability(self.DEFAULT_PROBABILITY)
+
+        self.assertEqual(probability, self.DEFAULT_PROBABILITY)
+
+    def test_boolean_metric_get_probability_too_large(self):
+        test_probability = 5
+
+        with self.assertRaises(Exception) as context:
+            BooleanMetric._get_probability(test_probability)
+            self.assertEqual(
+                context.exception,
+                Exception("Error: Please provide a float between 0 and 1 for probability."),
+            )
```

Comment:
you need to move assertEqual outside of the `with` otherwise it doesn't work (it will just pass even when the values are not equal)

and looks like it can't compare Exception directly, maybe something like
```python
        self.assertEqual(
            str(context.exception),
            "Error: Please provide a float between 0 and 1 for probability.",
        )
```